ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -913,7 +913,35 @@ public func zip<S: SequenceType, T, Error where S.Generator.Element == SignalPro
 	return .empty
 }
 
+extension SignalProducer where T: SignalProducerType, E == T.E {
+	/// Flattens the inner producers sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `producer` or an active inner producer emits an error, the returned
+	/// producer will forward that error immediately.
+	///
+	/// `Interrupted` events on inner producers will be treated like `Completed`
+	/// events on inner producers.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<T.T, E> {
```

Comment:
@neilpa what sort of issue did you have with compiler? In my case I just have an issues with types in ```lift``` call which looks like a legit call:
```
Cannot invoke 'lift' with an argument list of type '((Signal<SignalProducer<Self.T, Self.E>, Self.E>) -> Signal<Self.T, Self.E>)
```